File: client/models/CleaningChecklist.ts
export interface CleaningChecklist {
  checklistItemId: number;
  roomId: number;
  title: string;
  description?: string;
  isDefault: boolean;
}

export interface CleaningChecklistCreateRequest {
  room_id: number;
  title: string;
  description?: string;
  is_default?: boolean;
}

export interface CleaningChecklistUpdateRequest {
  title?: string;
  description?: string;
}

export interface CleaningCheckStatus {
  statusId: number;
  checklistItemId: number;
  membershipId: number;
  markedDate: string; // YYYY-MM-DD format
  isCompleted: boolean;
  updatedAt: string;
}

export interface CleaningCheckStatusCreateRequest {
  checklist_item_id: number;
  membership_id: number;
  marked_date: string; // YYYY-MM-DD format
  is_completed?: boolean;
}

export interface CleaningChecklistWithStatus {
  checklistItemId: number;
  roomId: number;
  title: string;
  description?: string;
  isDefault: boolean;
  assignedTo?: string;
  assignedMembershipIds?: string; // Comma-separated string of membership IDs
  isCompleted: boolean;
  statusId?: number;
}

File: api/src/models/cleaning_checklist.py
from datetime import datetime
from pydantic import BaseModel

class CleaningChecklist(BaseModel):
    checklist_item_id: int
    room_id: int
    title: str
    description: str | None = None
    is_default: bool

class CleaningChecklistCreateRequest(BaseModel):
    room_id: int
    title: str
    description: str | None = None
    is_default: bool = False

class CleaningChecklistUpdateRequest(BaseModel):
    title: str | None = None
    description: str | None = None

class CleaningCheckStatus(BaseModel):
    status_id: int
    checklist_item_id: int
    membership_id: int
    marked_date: datetime
    is_completed: bool
    updated_at: datetime

class CleaningCheckStatusCreateRequest(BaseModel):
    checklist_item_id: int
    membership_id: int
    marked_date: str  # YYYY-MM-DD format
    is_completed: bool = False

class CleaningCheckStatusUnassignRequest(BaseModel):
    checklist_item_id: int
    marked_date: str  # YYYY-MM-DD format

class CleaningChecklistWithStatus(BaseModel):
    checklist_item_id: int
    room_id: int
    title: str
    description: str | None = None
    is_default: bool
    assigned_to: str | None = None
    assigned_membership_ids: str | None = None
    is_completed: bool = False
    status_id: int | None = None
